apps/src/lib/kits/maker/WebSerialPortWrapper.js

```diff
@@ -24,67 +24,66 @@ export default class WebSerialPortWrapper extends EventEmitter {
     // TODO - not sure if this is used in Maker Toolkit yet
   }
 
-  /** Opens the serial port from Micro:Bit and starts reading from port.
-   * We are opening the port here and not reading from or writing to the port at this point in
-   * implementation. Later on when the entire Micro:WebSerial pathway is implemented,
-   * this function may be removed logic added for the MB in the open() function below,
-   * if appropriate.
-   */
-  openMBPort() {
+  // Opens the serial port from Micro:Bit and starts reading from port.
+  async openMBPort() {
```

Comment:
Actually since this function (and the one below) is now `async` it may be cleaner to use `await` rather than a promise chain, e.g.

```
async openMBPort() {
   ...
   try {
      await this.port.open();
      this.port.open = true;
      ...
      await this.readLoop();
      ...
      // return if needed
   } catch (error) {
       ...
   }
}
```

I also wonder if this would help resolve the situation where returning in one function works and doesn't in the other, or at least make finding the root cause easier...

Reply:
I tried using `await` in the way you laid out above, and it doesn't seem to work. I've tried many renditions of this to try to make this cleaner and easier to understand. I'm definitely open to other ways to implement!

Reply:
> Really interesting to see this come together! I am curious about the `return` vs no `return` in the open functions, but maybe we can look into that offline.

Yes - thanks for pairing and investigating this with me. Just documenting here that we added print statements and compared the execution of commands with `return` vs without `return` in the `openCPPort`. We saw a difference in the execution of commands for the CP which was not necessarily expected reading through the code.